Source/JavaScriptCore/builtins/JSIteratorConstructor.js

```diff
@@ -71,3 +71,47 @@ function from(value)
     // 6. Return wrapper.
     return @wrapForValidIteratorCreate(iterator, iteratorNextMethod);
 }
+
+// https://tc39.es/proposal-iterator-sequencing/#sec-iterator.concat
+function concat(/* ...items */)
+{
+    "use strict";
+
+    var count = @argumentCount();
```

Comment:
```suggestion
    var argumentCount = @argumentCount();
```

to avoid similarities with counters we pass to callback in other methods?